tests/main/prepare-image-validation-sets/task.yaml

```diff
@@ -0,0 +1,162 @@
+        fakestore new-snap-revision --dir "${STORE_DIR}" "$SNAP_PATH" --snap-rev-json snap-"$SNAP_NAME"-rev.json
+    }
+    
+    # Ensure the following snaps are available on the host snap to avoid the
+    # store script trying to fetch these after setting up the fake store.
+    echo Installing required snaps we will need
+    snap install core
+    snap install remarshal
+    snap install jq
+
+    echo Download snaps needed for the the model
+    snap download core
+    snap download pc --channel=20/stable
+    snap download pc-kernel --channel=20/stable
+    snap download core20
+    snap download snapd
+    
+    "$TESTSTOOLS"/store-state setup-fake-store "$STORE_DIR"
+    
+    # Sign the needed assertions for validation sets and model
+    gendeveloper1 sign-model < ./asserts/vs1.json > foo.assert
+    gendeveloper1 sign-model < ./asserts/vs2.json > bar.assert
+    gendeveloper1 sign-model < ./asserts/core-20.json > model.assert
+
+    echo Expose the needed assertions through the fakestore
+    cp "$TESTSLIB"/assertions/testrootorg-store.account-key "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR/asserts"
+    cp foo.assert "$STORE_DIR/asserts"
+    cp bar.assert "$STORE_DIR/asserts"
+
+    # It is not enough to copy the assertions, we must also ack them otherwise we
+    # will get an error about not being able to resolve the account key
+    snap ack "$STORE_DIR/asserts/testrootorg-store.account-key"
+    snap ack "$STORE_DIR/asserts/developer1.account"
+    snap ack "$STORE_DIR/asserts/developer1.account-key"
+    
+    # We now add all the required snaps to the fake store. It hardly matter which revision
+    # we give to them, as the fake store does not handle requests of specific revisions. Currently
+    # the fake-store will just return whatever revision there is.
+    echo "Adding snaps to the fake store"
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls core_*.snap)" 99T7MUlRhtI3U0QFgl5mXXESAiSwt776
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls pc_*.snap)" UqFziVZDHLSyO3TqSWgNBoAdHbLI4dAH
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls pc-kernel_*.snap)" pYVQrBcKmBa0mZ4CCN7ExT6jH8rY1hza
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls core20_*.snap)" DLqre5XGLbDqg9jPtiAhRRjDuPVa5X1q
+    "$TESTSTOOLS"/store-state make-snap-installable "$STORE_DIR" "$(ls snapd_*.snap)" PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4
+    install_snap_to_fakestore test-snapd-sh 13
+
+    # Use the fake-store to verify behaviour, so we can use our own assertions
+    export SNAPPY_FORCE_API_URL=http://$STORE_ADDR
+    echo Running prepare-image
+    snap prepare-image --channel edge --write-revisions --snap core --snap test-snapd-sh ./model.assert $ROOT
```

Comment:
shell check error? "$ROOT"

Reply:
Right, don't know why it wasn't caught, thanks for catching it